app/controllers/affiliates_controller.rb

```diff
@@ -1,18 +1,75 @@
 # frozen_string_literal: true
 
-class AffiliatesController < ApplicationController
+class AffiliatesController < Sellers::BaseController
   include Pagy::Backend
 
   PUBLIC_ACTIONS = %i[subscribe_posts unsubscribe_posts].freeze
-  before_action :authenticate_user!, except: PUBLIC_ACTIONS
+  skip_before_action :authenticate_user!, only: PUBLIC_ACTIONS
   after_action :verify_authorized, except: PUBLIC_ACTIONS
 
   before_action :set_direct_affiliate, only: PUBLIC_ACTIONS
-  before_action :set_meta, only: %i[index subscribe_posts unsubscribe_posts]
+  before_action :set_meta
   before_action :hide_layouts, only: PUBLIC_ACTIONS
 
+  layout "inertia", only: [:index, :onboarding, :new, :edit]
+
   def index
     authorize DirectAffiliate
+
+    page = params[:page]&.to_i || 1
+    query = params[:query]
+    sort_params = extract_sort_params
+
+    presenter = AffiliatesPresenter.new(
+      pundit_user,
+      page:,
+      query:,
+      sort: sort_params,
+      should_get_affiliate_requests: true
+    )
+    affiliates_data = presenter.index_props
+
+    if affiliates_data[:affiliates].empty? && affiliates_data[:affiliate_requests].empty? && page == 1 && query.blank?
+      redirect_to onboarding_affiliates_path and return
```

Comment:
Let's be consistent and prefer `&&` over `and`
```suggestion
      redirect_to onboarding_affiliates_path && return
```